src/decaf.cpp

```diff
@@ -195,6 +195,32 @@ namespace decaf {
     return ExecutionResult::Continue;
   }
 
+  ExecutionResult Interpreter::visitSDiv(llvm::BinaryOperator& op) {
+    StackFrame& frame = ctx->stack_top();
+
+    DECAF_ASSERT(op.getOperand(1) != 0);
+
+    auto lhs = normalize_to_int(frame.lookup(op.getOperand(0), *z3));
+    auto rhs = normalize_to_int(frame.lookup(op.getOperand(1), *z3));
+
+    frame.insert(&op, lhs / rhs);
+
+    return ExecutionResult::Continue;
+  }
+
+  ExecutionResult Interpreter::visitUDiv(llvm::BinaryOperator& op) {
+    StackFrame& frame = ctx->stack_top();
+
+    DECAF_ASSERT(op.getOperand(1) != 0);
+
+    auto lhs = normalize_to_uint(frame.lookup(op.getOperand(0), *z3));
+    auto rhs = normalize_to_uint(frame.lookup(op.getOperand(1), *z3));
+
```

Comment:
We need to have the solver check whether it's possible to get `rhs==0`.

This would look something like this:
```cpp
if (ctx->check(rhs == 0) == z3::sat) {
  queue->add_failure(ctx->solver.model());
}
ctx->add(rhs != 0);
```

Basically, we need to first check whether `rhs` could be 0. Then, if z3 finds a solution where it's possible, we need to add the test case. Finally, we assert that `rhs != 0`  and continue on.